### PR TITLE
feat(discover): convert time to timestamp.to_hour

### PR DESCRIPTION
**Pull request by alxu-sentry**
Some old queries still use "time". "timestamp.to_hour" has the same semantics,
so silently migrate them over.

### CODE REVIEW

File: src/sentry/migrations/0177_discover_time_to_timestamp_to_hour.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+from django.db import migrations
+from sentry.utils.query import RangeQuerySetWrapperWithProgressBar
+
+
+def time_to_timestamp_to_hour(apps, schema_editor):
+    DiscoverSavedQuery = apps.get_model("sentry", "DiscoverSavedQuery")
+    for query in RangeQuerySetWrapperWithProgressBar(DiscoverSavedQuery.objects.all()):
+        fields = query.query["fields"]
+        if "time" in fields:
```
Comment (wmak) [inline src/sentry/migrations/0177_discover_time_to_timestamp_to_hour.py:11]: Is "time" used in any query conditions (ie. `query.query["query"]`) that we need to update as well?

**Review by wmak — APPROVED** [review]
LGTM, I doubt time is in any conditions but we should check to be safe.

File: src/sentry/migrations/0177_discover_time_to_timestamp_to_hour.py (hunk 1 above)
Comment (alxu-sentry) [reply src/sentry/migrations/0177_discover_time_to_timestamp_to_hour.py:11]: Hm, somehow I totally forgot about query query query. I don't think anyone on sentry.io is using it but we should migrate it for self-hosted anyways.

**Review by wedamija — COMMENTED** [review]
Migration itself looks safe to me, I'll leave the rest of the review to @wmak

File: src/sentry/migrations/0177_discover_time_to_timestamp_to_hour.py (hunk 1 above)
Comment (alxu-sentry) [reply src/sentry/migrations/0177_discover_time_to_timestamp_to_hour.py:11]: There are some users on sentry.io, but writing a migration is complicated. Additionally, "time" will continue to work as before in queries, so migrating it isn't mandatory.

**Pull request closed without merging**